src/borg/testsuite/repo_new.py

```diff
@@ -8,6 +10,27 @@
 from .hashindex import H
 
 UNSPECIFIED = object()  # for default values where we can't use None
+EXCLUSIVE = True
+
+
+@pytest.fixture()
+def repository(tmpdir, exclusive=UNSPECIFIED, create=True):
+    if exclusive is UNSPECIFIED:
+        exclusive = EXCLUSIVE
+    repository_location = os.path.join(str(tmpdir), "repository")
+    repository = Repository(repository_location, exclusive=exclusive, create=create)
+    yield repository.__enter__()
+    repository.__exit__(None, None, None)
+    shutil.rmtree(str(tmpdir))
+
+
+@pytest.fixture()
+@contextmanager
+def reopened_repository(repository, exclusive=UNSPECIFIED):
+    repository.__exit__(None, None, None)
+    reopened_repository = Repository(repository.path, exclusive=exclusive, create=False)
+    yield reopened_repository.__enter__()
+    repository.__exit__(None, None, None)
```

Comment:
@RonnyPfannschmidt is this what you had in mind?

The only issue I run into now is that whenever a test uses this reopen, I get a warning that cleanup happened in `__del__`. Am I not exiting the repo correctly? Full warning below:

`Traceback (most recent call last):
    File "/Users/ted/Desktop/borg_workspace/borg/src/borg/repository.py", line 212, in __del__
      assert False, "cleanup happened in Repository.__del__"
  AssertionError: cleanup happened in Repository.__del__`
  
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))`
    
    @ThomasWaldmann @m3nu